game_view.h

```diff
@@ -37,9 +38,25 @@ class game_view
   ///Gets a const ref to m_game
   const game& get_game() const noexcept {return m_game; }
 
+  /// Get next view
+  view_mode get_next_view() const
+  {
+    return m_next_view;
+  }
+
+  // Get the size of the window
+  const sf::Vector2f get_window_size() { return m_window_size; }
+
+  // Get the window's state, for testing purposes only
+  bool is_window_open() { return m_window.isOpen(); }
```

Comment:
This is a usefull member function indeed. However the lack of const suggests that the view if modified just when you want to see if it open. Adding the const will make this member function even better.